### PR TITLE
Javascript capstone project

### PR DESCRIPTION
**Pull request by tylher**
Created a movie website that consumes a movie API and displays necessary information.

- [ ] Created a list of movies on the homepage
- [ ] Added comment and like functionalities to each item
- [ ] Wrote test for these functionalities
- [ ] Created a popup box to show details about a show or movie
- [ ] Styled the pages
- [ ] Wrote function to display the number of items, comments, and likes
- [ ] Documented the project in a Readme file

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+const searchShow = async (query) => {
+  const BASE_URL = `https://api.tvmaze.com/search/shows?q=${query}`;
+  await fetch(BASE_URL)
+    .then((respose) => respose.json())
+    .then((jsonData) => {
+      const results = jsonData.map((item) => item.show);
+      displayItem(results);
+      updateLikes(results);
+      const commentBtn = document.querySelectorAll('.comment-btn');
```
Comment (AnselemOdims) [inline src/index.js]: - [ ] In evaluating `Promises`, it is best practice to use either `async and await` or `thenable chain` but not both at the same time. If you remove the `async and await`, the code will still work. In this project you are required to use `async and await` , so please ensure to stick to the requirements.

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    .catch(() => {
+      displayItem([]);
+    });
+};
+
+const displayHomePage = () => {
+  const BASE_URL = 'https://api.tvmaze.com/shows';
+  fetch(BASE_URL)
+    .then((respose) => respose.json())
+    .then((jsonData) => {
```
Comment (AnselemOdims) [inline src/index.js]: - [ ] As pointed out earlier, please use `async and await` with the `fetch` method to send and receive data.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @team, 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please ensure to highlight a **piece of code** that you are proud of in the presentation video
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/index.js (hunk 1 above)
Comment (tylher) [reply src/index.js]: @AnselemOdims, thank you for the review. I am sorry, I do not understand what you mean, doesn't the async and await return a promise. Please how do I handle the promise?

Comment (AnselemOdims) [reply src/index.js]:
```
const response = await fetch(BASE_URL)
const data = await response.json()
const result = data.map((item) =>....)
```

Comment (tylher) [reply src/index.js]: Oh thank you very much

File: src/index.js (hunk 2 above)
Comment (tylher) [reply src/index.js]: We have done this , thank you  very much @AnselemOdims

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  await getShow(id).then((data) => {
+    popup.innerHTML = '';
+    popup.classList.remove('d-none');
+    popup.style.width = '80%';
+    popup.style.zIndex = '200';
+    renderPopUp(temp(data));
+    backgroundBlur();
+    displayComment(id);
+  });
+};
```
Comment (Gambit142) [inline src/index.js]: - [ ] Remember that you are to use the `async` and `await` with `fetch`. Just like the other reviewer said, using `async` and `await` with the promise `thenable chain` is not the best practice. You would want to either use `async` and `await` or the `thenable` chain for this project. One of the major requirements for this project is that you use async and await. Kindly implement this change.

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @tylher 👋 @joel-kalema 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

- For the records, this is one of the best Javascript Capstone Projects that I have seen. Great work team especially implementing the search bar feature 👏 


 ## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada:  :champagne:  :confetti_ball: 
Hi  :wave:   :wave: @tylher and @joel-kalema  

200w 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- No Linter errors :heavy_check_mark: 
- Your design is awesome :heavy_check_mark: 
- Your README file looks very professional :+1: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by tylher** [comment]
Awesome, thank you very much for the approval

**Pull request merged**